cdk/lib/index.ts

```diff
@@ -0,0 +1,302 @@
+import {
+  App,
+  CfnMapping,
+  Duration,
+  RemovalPolicy,
+  SecretValue,
+} from "@aws-cdk/core";
+import { Certificate } from "@aws-cdk/aws-certificatemanager";
+import type { GuStackProps } from "@guardian/cdk/lib/constructs/core/stack";
+import {
+  GuDnsRecordSet,
+  RecordType,
+} from "@guardian/cdk/lib/constructs/dns/dns-records";
+import { GuStack } from "@guardian/cdk/lib/constructs/core/stack";
+import { AccessScope, GuPlayApp } from "@guardian/cdk";
+import {
+  GuGetS3ObjectsPolicy,
+  GuPutCloudwatchMetricsPolicy,
+} from "@guardian/cdk/lib/constructs/iam";
+import { GuSecurityGroup, GuVpc } from "@guardian/cdk/lib/constructs/ec2";
+import { InstanceType, Port, SubnetType } from "@aws-cdk/aws-ec2";
+import { GuS3Bucket } from "@guardian/cdk/lib/constructs/s3";
+import {
+  AllowedMethods,
+  CacheCookieBehavior,
+  CacheHeaderBehavior,
+  CachePolicy,
+  CacheQueryStringBehavior,
+  Distribution,
+  OriginProtocolPolicy,
+} from "@aws-cdk/aws-cloudfront";
+import { LoadBalancerV2Origin } from "@aws-cdk/aws-cloudfront-origins";
+import {
+  Alarm,
+  ComparisonOperator,
+  Metric,
+  TreatMissingData,
+} from "@aws-cdk/aws-cloudwatch";
+import { GuDatabaseInstance } from "@guardian/cdk/lib/constructs/rds";
+import {
+  Credentials,
+  DatabaseInstanceEngine,
+  PostgresEngineVersion,
+  StorageType,
+  SubnetGroup,
+} from "@aws-cdk/aws-rds";
+import { GuArnParameter, GuParameter } from "@guardian/cdk/lib/constructs/core";
+
+export class Typerighter extends GuStack {
+  constructor(scope: App, id: string, props: GuStackProps) {
+    super(scope, id, props);
+
+    const parameters = {
+      MasterDBUsername: new GuParameter(this, "MasterDBUsername", {
+        description: "Master DB username",
+        default: "rule_manager",
+        type: "String",
+      }),
+      CheckerCertificate: new GuArnParameter(
+        this,
+        "CheckerCloudfrontCertificate",
+        {
+          description:
+            "The ARN of the certificate for the checker service Cloudfront distribution",
+        }
+      ),
+    };
+
+    const pandaAuthPolicy = new GuGetS3ObjectsPolicy(this, "PandaAuthPolicy", {
+      bucketName: "pan-domain-auth-settings",
+    });
+
+    const checkerDomainCODE = "checker.typerighter.code.dev-gutools.co.uk";
+    const checkerDomainPROD = "checker.typerighter.gutools.co.uk";
+    const managerDomainCODE = "manager.typerighter.code.dev-gutools.co.uk";
+    const managerDomainPROD = "manager.typerighter.gutools.co.uk";
+
+    const stageLookup = new CfnMapping(this, "LowercaseStageLookup", {
```

Comment:
Could we use [`withStageDependentValue`](https://github.com/guardian/cdk/blob/aadfe79af61bed401393e658facdb5e1eba22706/src/constructs/core/stack.ts#L77-L87) from `GuStack` here?

Reply:
We can! And we will! Thanks for making me aware of this very useful API.